text/0000-prepare-operation.md

```diff
@@ -0,0 +1,387 @@
+# Meta
+[meta]: #meta
+- Name: Prepare Operation
+- Start Date: 2022-02-07
+- Author(s): @jromero, [@david-caro](https://github.com/david-caro)
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+It is common for platform to execute common operations before executing the lifecycle. These sort of operations include, downloading buildpacks, making changes the the `order.toml`, cleaning up the workspace, and more. These operations are typically influenced by configuration. One of those common configuration inputs is the `project.toml` (aka Project Descriptor). This RFC, proposes a contractual interface for a Prepare operation that is executed before the Build operation. The benefits for doing so are it enables common functionality to be made portable. Additionally, it allows the project to provide guidance on what the expected behaviour is regarding input configuration such as the `project.toml`.
+
+# Definitions
+[definitions]: #definitions
+
+- namespace - A table within `project.toml` that encompasses a set of properties.
+- Prepare operation - a new operation that occurs before the Build operation.
+- `project.toml` - A project descriptor file specified [here][project-descriptor-spec].
+
+[project-descriptor-spec]: https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md
+
+# Motivation
+[motivation]: #motivation
+
+The motivation for this change was to enable the following goals from being achieved:
+
+  - **Goal 1: Serializing CLI configuration**
+    > As a user, I would like to be able to serialize and share the parameters I use with certain platforms such as `pack`.
+  - **Goal 2: A recognizable file in repositories**
+    > As a user, I would like to be able to recognize, based on the file system, if a project is using Cloud Native Buildpacks.
```

Comment:
I'm not sure I understand how this goal fits in

Reply:
Is it because I can not depend on `project.toml` to work, regardless of platform?

Reply:
I think so yes. The project descriptor file currently can be ignored by the platform in any case right?
This adds the restriction that now they `MUST` apply at least the `io.buildpacks` section, making the combo (`project.toml` + `io.buildpacks`) a way to identify if the project uses buildpacks, regardless of the platform.

So summarizing my own words, this RFC ensures that if there's a `project.toml`, and it has a `io.buildpacks' section, then you can rely on it using those buildpacks regardless of platform.

Does that make sense? (/me is still trying to understand both the original motivation and current flows, so please be extra verbose with the replies :pray: )